pallets/gear/src/lib.rs

```diff
@@ -1673,7 +1673,7 @@ pub mod pallet {
             // First we reserve enough funds on the account to pay for `gas_limit`
             // and to transfer declared value.
             <T as Config>::Currency::reserve(&who, reserve_fee + value)
-                .map_err(|_| Error::<T>::NotEnoughBalanceForReserve)?;
+                .defensive_map_err(|_| Error::<T>::NotEnoughBalanceForReserve)?;
```

Comment:
Please add comments why do they unreachable:
- pre-validation on tx pool insertion
- validation on tx pool removal to be included in block